### PR TITLE
Support Building on OpenBSD

### PR DESCRIPTION
**Pull request by crypto-ape**
Hey Monkeys!

I had some interest in being able to run a BitShares node on OpenBSD.

This PR with [this](https://github.com/bitshares/bitshares-core/pull/1910) bitshares-core PR contains changes allowing to successfully build the code base under OpenBSD.

The work is not done yet, because, at least for me, the compiled executables crash (with a segmentation fault) on calling some boost functions (hastily done some call tracing).

I plan to investigate further. So far, I wanted to share and close the work already done.

Ape out!

### CODE REVIEW

File: src/crypto/elliptic_common.cpp [hunk 1]
```diff
@@ -231,12 +229,11 @@ namespace fc { namespace ecc {
...
     static std::string _to_base58( const extended_key_data& key )
     {
-        size_t buf_len = key.size() + 4;
-        char *buffer = (char*)alloca(buf_len);
+        char buffer[key.size() + 4]; // it's a small static array => allocate on stack
         memcpy( buffer, key.begin(), key.size() );
         fc::sha256 double_hash = fc::sha256::hash( fc::sha256::hash( key.begin(), key.size() ));
         memcpy( buffer + key.size(), double_hash.data(), 4 );
-        return fc::to_base58( buffer, buf_len );
+        return fc::to_base58( buffer, key.size() );
```
Comment (pmconrad) [inline src/crypto/elliptic_common.cpp]: This looks wrong. `buf_len` used to be equal to `key.size()+4`.

**Review by pmconrad — CHANGES_REQUESTED** [review]

**Comment by pmconrad** [comment]
Please rebase on latest master.

File: src/crypto/elliptic_common.cpp (hunk 1 above)
Comment (crypto-ape) [reply src/crypto/elliptic_common.cpp]: Sure, it should be the local buffer size. Poor me, such a mistake.

__Thanks!__

**Review by pmconrad — APPROVED** [review]

**Comment by pmconrad** [comment]
Haven't tested on OpenBSD, but looking good. Thanks!

File: src/variant.cpp [hunk 2]
```diff
@@ -701,7 +701,9 @@ void from_variant( const variant& var, uint128_t& vo, uint32_t max_depth )
 #endif
 }
 
-#ifdef __APPLE__
+#if defined(__APPLE__)
+#elif defined(__OpenBSD__)
+   void to_variant( size_t s, variant& v, uint32_t max_depth ) { v = variant( int64_t(s) ); }
```
Comment (abitmore) [inline src/variant.cpp:706]: Not sure whether this is really needed. I suspect it's unused since macOS build doesn't rely on it.

Comment (abitmore) [reply src/variant.cpp:706]: OK, found the code for macOS in `variant.hpp` implemented with `inline`.

Comment (abitmore) [reply src/variant.cpp:706]: Using `int64_t` but not `uint64_t` here is probably wrong.

Comment (abitmore) [reply src/variant.cpp:706]: Fix: https://github.com/bitshares/bitshares-fc/pull/194.

**Pull request merged**